src/GraphControl/Control/Grapher.h

```diff
@@ -155,6 +156,105 @@ public
         void PlotGraph();
         void AnalyzeEquation(GraphControl::Equation ^ equation);
 
+        // We can't use the EvalTrigUnitMode enum directly in as the property type because it comes from another module which doesn't expose
+        // it as a public enum class.  So the compiler doesn't recognize it as a valid type for the ABI boundary.
+        property int TrigUnitMode
+        {
+            void set(int value)
+            {
+                if (value != (int)m_solver->EvalOptions().GetTrigUnitMode())
+                {
+                    m_solver->EvalOptions().SetTrigUnitMode((EvalTrigUnitMode)value);
+                    PlotGraph();
+                }
+            }
+
+            int get()
+            {
+                return (int)m_solver->EvalOptions().GetTrigUnitMode();
+            }
+        }
+
+        property double XAxisMin
+        {
+            double get()
+            {
+                return m_graph->GetOptions().GetDefaultXRange().first;
+            }
+            void set(double value)
+            {
+                std::pair<double, double> newValue(value, XAxisMax);
```

Comment:
Not directly related to this PR but it makes reproing a crash easier when the axis is set to very large/small numbers

Try setting any of the y/x max and mins to very large or very low numbers, then hover your mouse over a graphed line near those large numbers.

in RenderMain::Render() the call to renderer->GetClosePointData crashes with 'An invalid parameter was passed to a function that considers invalid parameters fatal.'

I think it crashes because the range is a double but the API takes a float for the xValueOut and yValueOut params. We may want to file a bug, but also may want to consider handling the ranges as doubles and limiting their size if we can't get a bug fix.

Reply:
Actually I don't repro this outside of this branch, so maybe something in this branch caused the crash?

Reply:
👀

Reply:
Yes, we should file a bug, 
`        HRESULT GetClosePointData(float inScreenPointX, float inScreenPointY, int& formulaIdOut, float& xScreenPointOut, float& yScreenPointOut, float& xValueOut, float& yValueOut) override`

It should use double, not float (all other fields use double (xmin, xmax, etc...)

Reply:
In the meantime, we should ignore this exception

Reply:
Unfortunately, we can't try... catch this exception, it happens on another thread. I'm able to repro the issue with the zoom out function, you need to have an equation and zoom out up to 1e17